src/sidebar/components/SidebarContent.js

```diff
@@ -0,0 +1,35 @@
+import classnames from 'classnames';
+
+import { downcastRef } from '../util/typing';
+
+/**
+ * @typedef {import('../../types/sidebar').PresentationalDivComponentProps} SidebarContentProps
+ */
+
+/**
+ * Render "page"-like content in a container that centers it and constrains
+ * it to a maximum content width.
+ *
+ * @param {SidebarContentProps} props
+ */
+export default function SidebarContent({
```

Comment:
Hmm... I do have misgivings about creating a component for something that could just be a CSS class, since there is overhead for adding more depth to the component tree. I don't have concrete numbers to hand. This would mainly be an issue in scenarios where a lot of such components are created. I realize we need some concrete guidelines around this to move forwards, maybe of a "use in moderation" nature.

Aside from performance there is also quite a bit of boilerplate here. This could potentially be wrapped up in a utility, which would also make it easy to identify these components which exist purely as styling wrappers.

```diff
diff --git a/src/sidebar/components/SidebarContent.js b/src/sidebar/components/SidebarContent.js
index 3795e6bc2..61a179e78 100644
--- a/src/sidebar/components/SidebarContent.js
+++ b/src/sidebar/components/SidebarContent.js
@@ -3,33 +3,52 @@ import classnames from 'classnames';
 import { downcastRef } from '../util/typing';
 
 /**
- * @typedef {import('../../types/sidebar').PresentationalDivComponentProps} SidebarContentProps
+ * @typedef {import('preact').JSX.HTMLAttributes<HTMLDivElement>} HTMLAttributes
  */
 
 /**
- * Render "page"-like content in a container that centers it and constrains
- * it to a maximum content width.
+ * @typedef PresentationalComponentProps
+ * @prop {import('preact').ComponentChildren} [children]
+ * @prop {string|string[]} [classes]
+ * @prop {import('preact').Ref<HTMLElement>} [elementRef]
+ */
+
+/**
+ * Make a presentational component which wraps children to apply layout and
+ * styling.
  *
- * @param {SidebarContentProps} props
+ * TODO: Move this to a utility module.
+ *
+ * @param {string} displayName
+ * @param {string|string[]} classes
  */
-export default function SidebarContent({
-  children,
-  classes,
-  elementRef,
-  ...restProps
-}) {
-  return (
-    <div
-      className={classnames(
-        // Center this content (auto margins). For larger viewports, set a
-        // maximum width (768px) and add some horizontal padding.
-        'mx-auto lg:px-16 lg:max-w-3xl',
-        classes
-      )}
-      {...restProps}
-      ref={downcastRef(elementRef)}
-    >
-      {children}
-    </div>
-  );
+export function makePresentationalComponent(displayName, classes) {
+  /**
+   * @param {HTMLAttributes & PresentationalComponentProps} props
+   */
+  function PresentationalComponent({
+    classes: extraClasses,
+    elementRef,
+    children,
+    ...restProps
+  }) {
+    return (
+      <div
+        className={classnames(classes, extraClasses)}
+        ref={downcastRef(elementRef)}
+        {...restProps}
+      >
+        {children}
+      </div>
+    );
+  }
+  PresentationalComponent.displayName = displayName;
+  return PresentationalComponent;
 }
+
+export default makePresentationalComponent(
+  'SidebarContent',
+  // Center this content (auto margins). For larger viewports, set a
+  // maximum width (768px) and add some horizontal padding.
+  'mx-auto lg:px-16 lg:max-w-3xl'
+);
```